vhost-device-sound/src/stream.rs

```diff
@@ -362,7 +376,7 @@ mod tests {
 
         let desc_out = Descriptor::new(
             next_addr,
-            std::mem::size_of::<R>() as u32,
+            std::mem::size_of::<R>() as u32 + payload_len,
```

Comment:
I did not check, but I assume the Descriptor takes a usize/u64 here? So maybe this should both convert to u64 before summing to avoid overflows?

Reply:
Descriptor takes u32 as len.

Reply:
yeah, I agree that we should check overflows

Reply:
This is part of the tests in which overflow is not possible for the moment. Do you think I should check for overflow anyway?

Reply:
What would you think about this?
```suggestion
            (std::mem::size_of::<R>() as u32).saturating_add(payload_len),
```

Reply:
Nope, maybe something more like `(std::mem::size_of::<R>() as u32).checked_add(payload_len).unwrpa()`, so it will panic if the overflow will happen.

Because a `Descriptor` can contain only a buffer where the length is on 32 bit.